roofit/roofitcore/src/RooAbsTestStatistic.cxx

```diff
@@ -507,9 +507,7 @@ void RooAbsTestStatistic::initSimMode(RooSimultaneous* simpdf, RooAbsData* data,
         binnedL = true ;
       } else if (pdf->IsA()->InheritsFrom(RooProdPdf::Class())) {
         // Default case: top-level pdf is a product of RRSP and other pdfs
-        RooFIter iter = ((RooProdPdf*)pdf)->pdfList().fwdIterator() ;
-        RooAbsArg* component ;
-        while ((component = iter.next())) {
+        for (auto const* component : ((RooProdPdf*)pdf)->pdfList()) {
```

Comment:
```suggestion
        for (auto const* component : static_cast<RooProdPdf*>(pdf)->pdfList()) {
```
When you are already touching the code, you should replace C-style casts with C++ casts please